ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgoCommonUtils.scala

```diff
@@ -52,6 +53,11 @@ trait H2OAlgoCommonUtils extends EstimatorCommonUtils {
     }
 
     val featureColumns = getFeaturesCols().map(sanitize).map(col)
+
+    if (dataset.select(featureColumns: _*).distinct().count() == 1) {
```

Comment:
Isn't [this H2O option](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/algo-params/ignore_const_cols.html#ignore-const-cols) important for make the condition?

Reply:
Good find! But I don't think so; we just need to check whether all columns are constants. If this option is applied, H2O will have no columns to use as features and we throw the same exception which I thing is correct. WDYT?

Reply:
It seems that you right. The comment inside H2O code says:
`Check that at least some columns are not-constant and not-all-NAs`